### PR TITLE
feat: adds email notification when verified badge added

### LINKED ISSUE #2688: [feature] badges add transactional email notifications for users and admins
Precious plastic to provide listing of required transactional emails, their trigger and the email content.

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Screenshot 2023-09-27 at 19 29 43

## Git Issues

Closes #2688

### CODE REVIEW

File: functions/src/emailNotifications/verifiedBadgeEmails.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as functions from 'firebase-functions'
+import { db } from '../Firebase/firestoreDB'
+import { getUserEmail } from './utils'
+import { DB_ENDPOINTS } from 'oa-shared'
+import { getUserVerifiedBadgeAddedEmail } from './templates'
+import type { IUserDB } from '../models'
+import { withErrorAlerting } from '../alerting/errorAlerting'
+import type { QueryDocumentSnapshot } from 'firebase-admin/firestore'
+
+exports.handleUserVerifiedBadgeUpdate = functions.firestore
```
Comment (evakill) [inline functions/src/emailNotifications/verifiedBadgeEmails.ts:10]: we can also use ES6 exports for these, is there a reason you prefer `exports.` here?

File: functions/src/emailNotifications/verifiedBadgeEmails.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  )
+
+const handleUserVerifiedBadgeChange = async (
+  change: functions.Change<QueryDocumentSnapshot<IUserDB>>,
+) => {
+  const after = change.after.exists ? change.after.data() : null
+  const before = change.before.exists ? change.before.data() : null
+  const toUserEmail = after ? await getUserEmail(after._authID) : null
+
+  if (before.badges?.verified === false && after.badges?.verified === true) {
```
Comment (evakill) [inline functions/src/emailNotifications/verifiedBadgeEmails.ts:23]: we should also send if `before.badges?.verified` is undefined, right?

**Review by evakill — COMMENTED** [review]
eek sorry about the merge conflicts

File: functions/src/emailNotifications/verifiedBadgeEmails.ts (hunk 1 above)
Comment (thisislawatts) [reply functions/src/emailNotifications/verifiedBadgeEmails.ts]: Fixed here: https://github.com/ONEARMY/community-platform/pull/2820/commits/2929bc6414493ade2b72e83aadabd9c2aab6744c

No preference from myside

File: functions/src/emailNotifications/verifiedBadgeEmails.ts (hunk 2 above)
Comment (thisislawatts) [reply functions/src/emailNotifications/verifiedBadgeEmails.ts]: Addressed in https://github.com/ONEARMY/community-platform/pull/2820/commits/4ce433a59a692e78237a6ab4cb7570b347fb26ae

**Review by evakill — APPROVED** [review]
lgtm!

**Pull request merged**